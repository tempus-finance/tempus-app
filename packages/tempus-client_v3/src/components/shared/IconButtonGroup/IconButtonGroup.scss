.tc__iconButtonGroup {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 40px;

  button {
    &:not(:last-child) {
      margin-right: 10px;
    }
  }
}
