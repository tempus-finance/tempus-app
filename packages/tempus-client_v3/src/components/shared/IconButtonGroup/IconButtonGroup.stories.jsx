import IconButtonGroup from './IconButtonGroup';

export default {
  title: 'IconButtonGroup',
  component: IconButtonGroup,
  argTypes: {
    types: {
      control: {
        type: 'array',
      },
    },
    selected: {
      control: {
        type: 'boolean',
      },
    },
  },
};

const style = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  height: '100px',
};

const Template = args => (
  <div style={style}>
    <IconButtonGroup {...args} />
  </div>
);

export const Primary = Template.bind({});
Primary.args = {
  types: ['grid-view', 'list-view'],
};
