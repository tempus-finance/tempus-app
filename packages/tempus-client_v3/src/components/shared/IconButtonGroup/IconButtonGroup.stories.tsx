import { ComponentStory } from '@storybook/react';
import IconButtonGroup from './IconButtonGroup';

export default {
  title: 'IconButtonGroup',
  component: IconButtonGroup,
  argTypes: {
    types: {
      control: {
        type: 'array',
      },
    },
  },
};

const style = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  height: '100px',
};

const Template: ComponentStory<typeof IconButtonGroup> = args => (
  <div style={style}>
    <IconButtonGroup {...args} />
  </div>
);

export const Primary = Template.bind({});
Primary.args = {
  types: ['grid-view', 'list-view'],
  onChange: () => {},
};
