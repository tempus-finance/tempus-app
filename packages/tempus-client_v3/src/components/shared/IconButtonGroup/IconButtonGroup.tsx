import { FC, useState } from 'react';
import { IconType } from '../Icon/Icon';
import IconButton from '../IconButton/IconButton';
import './IconButtonGroup.scss';

interface IconButtonGroupProps {
  types: IconType[];
  onChange: (selected: IconType) => void;
}

const IconButtonGroup: FC<IconButtonGroupProps> = props => {
  const { types, onChange } = props;

  const [selectedType, setSelectedType] = useState<IconType>(types[0]);

  const onClick = (value: IconType) => {
    setSelectedType(value);

    onChange(selectedType);
  };

  return (
    <div className="tc__iconButtonGroup">
      {types.map(type => {
        return <IconButton key={type} type={type} onClick={onClick} selected={type === selectedType} />;
      })}
    </div>
  );
};
export default IconButtonGroup;
