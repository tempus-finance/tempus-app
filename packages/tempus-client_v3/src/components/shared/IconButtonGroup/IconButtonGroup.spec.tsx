import { fireEvent, render } from '@testing-library/react';
import IconButtonGroup, { IconButtonGroupProps } from './IconButtonGroup';

const mockOnChange = jest.fn();

const defaultProps: IconButtonGroupProps = {
  types: ['grid-view', 'list-view'],
  onChange: mockOnChange,
};

const subject = (props: IconButtonGroupProps) => render(<IconButtonGroup {...props} />);

describe('IconButtonGroup', () => {
  it('renders a group of icon buttons', () => {
    const { container } = subject(defaultProps);

    const actualIconButtonGroup = container.querySelector('.tc__iconButtonGroup');

    expect(actualIconButtonGroup).not.toBeNull();

    expect(actualIconButtonGroup).toMatchSnapshot();
  });

  it('calls `onClick` when an icon button is clicked', () => {
    const { getAllByRole } = subject(defaultProps);

    const [firstButton, secondButton] = getAllByRole('button');

    fireEvent.click(secondButton);
    expect(mockOnChange).toHaveBeenCalledWith('list-view');

    fireEvent.click(firstButton);
    expect(mockOnChange).toHaveBeenCalledWith('grid-view');

    expect(mockOnChange).toHaveBeenCalledTimes(2);
  });
});
