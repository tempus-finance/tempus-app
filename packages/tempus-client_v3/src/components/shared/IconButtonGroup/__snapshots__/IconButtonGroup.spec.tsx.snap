// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonGroup renders a group of icon buttons 1`] = `
<div
  class="tc__iconButtonGroup"
>
  <button
    class="tc__btn tc__iconButton__selected"
    data-selected="false"
  >
    <svg
      class="tc__icon tc__icon-grid-view"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M1.2 0C0.537258 0 0 0.537259 0 1.2V11.4H11.4V0H1.2ZM12.6 0V11.4H24V1.2C24 0.537258 23.4627 0 22.8 0H12.6ZM24 12.6H12.6V24H22.8C23.4627 24 24 23.4627 24 22.8V12.6ZM11.4 24V12.6H0V22.8C0 23.4627 0.537259 24 1.2 24H11.4Z"
        fill="#FFFFFF"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <button
    class="tc__btn tc__iconButton__deselected"
    data-selected="false"
  >
    <svg
      class="tc__icon tc__icon-list-view"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6.6 3H23.4C23.7314 3 24 3.26863 24 3.6V6C24 6.33137 23.7314 6.6 23.4 6.6H6.6V3Z"
        fill="#BFC0C0"
      />
      <path
        d="M0 3.6C0 3.26863 0.268629 3 0.6 3H5.4V6.6H0.6C0.268629 6.6 0 6.33137 0 6V3.6Z"
        fill="#BFC0C0"
      />
      <path
        d="M6.6 10.2H23.4C23.7314 10.2 24 10.4686 24 10.8V13.2C24 13.5314 23.7314 13.8 23.4 13.8H6.6V10.2Z"
        fill="#BFC0C0"
      />
      <path
        d="M0 10.8C0 10.4686 0.268629 10.2 0.6 10.2H5.4V13.8H0.6C0.268629 13.8 0 13.5314 0 13.2V10.8Z"
        fill="#BFC0C0"
      />
      <path
        d="M6.6 17.4H23.4C23.7314 17.4 24 17.6686 24 18V20.4C24 20.7314 23.7314 21 23.4 21H6.6V17.4Z"
        fill="#BFC0C0"
      />
      <path
        d="M0 18C0 17.6686 0.268629 17.4 0.6 17.4H5.4V21H0.6C0.268629 21 0 20.7314 0 20.4V18Z"
        fill="#BFC0C0"
      />
    </svg>
  </button>
</div>
`;
